tests/test_runner/test_checkpoint.py

```diff
@@ -221,6 +221,76 @@ def __init__(self):
     os.remove(checkpoint_path)
 
 
+def test_load_checkpoint_metadata():
+    import os
+    import tempfile
+
+    from mmcv.runner import load_checkpoint, save_checkpoint
+
+    class ModelV1(nn.Module):
+
+        def __init__(self):
+            super().__init__()
+            self.block = Block()
+            self.conv1 = nn.Conv2d(3, 3, 1)
+            self.conv2 = nn.Conv2d(3, 3, 1)
+            nn.init.normal_(self.conv1.weight)
+            nn.init.normal_(self.conv2.weight)
+
+    class ModelV2(nn.Module):
+        _version = 2
+
+        def __init__(self):
+            super().__init__()
+            self.block = Block()
+            self.conv0 = nn.Conv2d(3, 3, 1)
+            self.conv1 = nn.Conv2d(3, 3, 1)
+            nn.init.normal_(self.conv0.weight)
+            nn.init.normal_(self.conv1.weight)
+
+        def _load_from_state_dict(self, state_dict, prefix, local_metadata,
+                                  *args, **kwargs):
+            """load checkpoints."""
+
+            # Names of some parameters in has been changed.
+            version = local_metadata.get('version', None)
+            if version is None or version < 2:
+                state_dict_keys = list(state_dict.keys())
+                convert_map = {'conv1': 'conv0', 'conv2': 'conv1'}
+                for k in state_dict_keys:
+                    for ori_str, new_str in convert_map.items():
+                        if k.startswith(prefix + ori_str):
+                            new_key = k.replace(ori_str, new_str)
+                            state_dict[new_key] = state_dict[k]
+                            del state_dict[k]
```

Comment:
```suggestion
                new_state_dict = {}
                key_mappings = {'conv1': 'conv0', 'conv2': 'conv1'}
                for k in state_dict.keys():
                    for ori_str, new_str in key_mappings.items():
                        if k.startswith(prefix + ori_str):
                            new_key = k.replace(ori_str, new_str)
                            new_state_dict[new_key] = state_dict[k]
                            break
                    else:
                        new_state_dict[k] = state_dict[k]
```

Reply:
Perhaps we cannot apply it.
https://github.com/open-mmlab/mmcv/blob/d3b057240829a0662125d5e83e002b33cd3fa874/mmcv/runner/checkpoint.py#L66-L79
When loading the checkpoint, we recursively call `_load_from_state_dict` from each module. If we don't inplace modify the state_dict, this modification will be ineffective.

Reply:
got it